Contest and submission model #21

jhanvi811 opened the pull request:
- fixed issue
- contest model
- submission model

Migration command
- flask run
- flask db init
- flask db migrate
- flask db upgrade

Commit:
contest_model

Models.py file consist of following:
1. Database connection
2. Migration
3. Contest_user
4. Contest_table

Commit:
New submission

- fixed issue
- contest model
- submission model

Migration command
- flask run
- flask db init
- flask db migrate
- flask db upgrade

bonnieli commented on server/app.py:
```diff
@@ -1,11 +1,29 @@
 
+POSTGRES = {
+    'user': 'postgres',
+    'pw': 'Hema@101',
```
we need to hide all this information in environment variables

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+from sqlalchemy.sql import func
+from sqlalchemy import ForeignKey
+
+class contest_user(db.Model):
```
for classes, we should be naming them in PascalCase

```suggestion
class User(db.Model):
```

bonnieli replied:
all Users should be the same so this may not be required

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String())
+    password = db.Column(db.String())
+    createdTime = db.Column(db.DateTime, default=func.utcnow())
```
would be best if these attributes are snake_case

```suggestion
    created_time = db.Column(db.DateTime, default=func.utcnow())
```

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+
+    id = db.Column(db.Integer, primary_key=True)
+    sub_email = db.Column(db.String())
+    sub_password = db.Column(db.String())
```
some inconsistency here, i recommend using snake_case for attributes

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+        self.password = password
+        self.createdTime = createdTime
+
+class submission_user(db.Model):
```
What is this model used for? what would be the difference between a submission user and a regular user?

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+    sub_password = db.Column(db.String())
+    createdTime = db.Column(db.DateTime, default=func.utcnow())
+
+class contest_table(db.Model):
```
```suggestion
class Contest(db.Model):
```

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+        self.updateTime = updateTime
+        self.contestCreater = contestCreater
+
+class submission_table(db.Model):
```
```suggestion
class Submission(db.Model):
```

bonnieli commented:
You might not be able to address everything because the User model will naturally be created in the authentication feature

bonnieli commented:
This PR is also setting up the db

bonnieli commented on server/app.py:
```diff
@@ -1,11 +1,29 @@
+db = SQLAlchemy(app)
+migrate = Migrate(app, db)
+
+import models
```
i don't think you're using this atm

edyhtan commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+class contest_user(db.Model):
+    __tablename__ = 'contest_user'
+
+    id = db.Column(db.Integer, primary_key=True)
```
id is built-in function name for python. 

Let's use "username" for consistency.

jhanvi811 replied:
Id is used as a reference in other tables, that's why I used built-in function(when row creates, id is Automatically assigned to row) so in future conflict doesn't happen.

edyhtan replied:
I see what you mean. 
Then we should also include username as a col since that's the only way users are able to identify themselves during login.

jhanvi811 replied:
that's a good idea. I am just gonna apply.

Simeon10 commented on server/models.py:
```diff
@@ -0,0 +1,72 @@
+    contestCreater = db.Column(db.Integer, ForeignKey('contest_user.id'))
+
+    def __init__(self, id, title, description, prizeContest, deadlineDate, createdTime, updateTime, contestCreater):
+        self.id = id
```
I don't think "id" should be one of the parameters when initializing a new object because isn't that auto generated?
Or am I misunderstanding this?

bonnieli replied:
you're right, this should be required to init a model - it should be auto generated and handled via Postgres

Commit:
changes applied

Added username to user table and @bonnieli suggestion are applied

bonnieli commented on server/.env:
```diff
@@ -1,2 +1,8 @@
\ No newline at end of file
+FLASK_ENV="development"
+
+export DB_USER = "postgres"
```
you want to git ignore this file now so your password doesn't get shared

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,65 @@
+        self.update_time = update_time
+
+
+
```
too many extra lines

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,65 @@
+    update_time = db.Column(db.DateTime, onupdate=func.utcnow()) # store updated time of row
+    contest_creater = db.Column(db.Integer, ForeignKey('contest_user.id'))
+
+    def __init__(self, id, title, description, prize_contest, deadline_date, created_time, update_time, contest_creater):
```
created_time should also get autogenerated on init and won't be required to pass in

bonnieli commented on server/models.py:
```diff
@@ -0,0 +1,65 @@
+    deadline_date = db.Column(db.DateTime)
+    created_time = db.Column(db.DateTime, default=func.utcnow())
+    update_time = db.Column(db.DateTime, onupdate=func.utcnow()) # store updated time of row
+    contest_creater = db.Column(db.Integer, ForeignKey('contest_user.id'))
```
```suggestion
    contest_creater = db.Column(db.Integer, ForeignKey('user.id'))
```

Commit:
migration files

added migration files

bonnieli commented on server/migrations/versions/f94e1dc7996d_.py:
```diff
@@ -0,0 +1,61 @@
+    sa.PrimaryKeyConstraint('id')
+    )
+    op.create_table('submission',
+    sa.Column('id', sa.Integer(), nullable=False),
```
for all IDs, you probably want to make them autoincrement too 
```
sa.Column('id', sa.Ingeter(), autoincrement=True, nullable=False)
```

jhanvi811 replied:
suggestion applied.

Commit:
changes in migration

autoincrement=True in all IDs

Commit:
resolved conflict in .env file

Commit:
changes in .env and app.py

changes in .env and app.py

Commit:
models.py changes for id and created_time

- id column  : autoincrement=True, nullable=False
- created_time deleted from __init __ methods.

bonnieli committed:
Merge branch 'dev' into contestAndSubmission-model-#21
